### PR TITLE
Improve Indicating Errors docs

### PR DESCRIPTION
**Pull request by beckermarc**
Better explain how targets with SAP Fiori work.
Addresses https://github.tools.sap/cap/issues/issues/16273

### CODE REVIEW

File: java/event-handlers/indicating-errors.md [hunk 1]
```diff
@@ -217,11 +223,15 @@ public void validateTitle(CdsCreateEventContext context, Books book) {
...
+    // which is the same as using plain string
+    // (assuming direct POST request)
     throw new ServiceException(ErrorStatuses.BAD_REQUEST, "No title specified")
         .messageTarget("title");
 
-    // ...
+    // which is the same as using plain string
+    // (assuming surrounding bound action request, e.g. draftActivate)
+    throw new ServiceException(ErrorStatuses.BAD_REQUEST, "No title specified")
+        .messageTarget("in/title");
```
[the remark this answers is not in the document]
Comment (beckermarc) [reply java/event-handlers/indicating-errors.md:234]: I wouldn't mention it here, as this code refers to the explicit example model from above, where the name is not overriden. The text above doesn't state that the binding parameter is always called `in`.

[the remark this answers is not in the document]
Comment (bugwelle) [reply java/event-handlers/indicating-errors.md:234]: Yes, but the surrounding text (and I mean the same sectino), also doesn't mention "binding parameter". Users will only see the magic string "in/".

[the remark this answers is not in the document]
Comment (beckermarc) [reply java/event-handlers/indicating-errors.md:235]: It's explained in the text above that the binding parameter needs to be prefixed. I however mentioned the term binding parameter in the comment now here again.

[the remark this answers is not in the document]
Comment (bugwelle) [reply java/event-handlers/indicating-errors.md:235]: Thanks! :)

**Review by bugwelle — APPROVED** [review]

**Comment by beckermarc** [comment]
@BraunMatthias Do you also want to review or could we continue with review from @smahati and @renejeglinsky?

**Review by BraunMatthias — APPROVED** [review]

File: java/event-handlers/indicating-errors.md [hunk 2]
```diff
@@ -150,11 +150,18 @@ When SAP Fiori interprets messages it can handle an additional `target` property
 When specifying messages in the `sap-messages` HTTP header, SAP Fiori mostly ignores the `target` value.
 Therefore, specifying the `target` can only correctly be used when throwing a `ServiceException` as SAP Fiori correctly handles the `target` property in OData V4 error responses.
 
-A message target is always relative to an input parameter in the event context. For CRUD-based events this is usually the `cqn` parameter you can find in the underlying map of the event context. For action or function events you find their input parameters in the map, as well.
-
-Therefore, when creating a message target, one of these event context parameters needs to be selected to specify what the relative message target path refers to.
+A message target is always relative to an input parameter in the event context.
+For CRUD-based events this is always the `cqn` parameter, which represents and carries the payload of the request.
+For actions or functions a message target can either be relative to the entity the action or function is bound to (represented by the `cqn` parameter) or relative to a parameter of the action or function.
```
Comment (smahati) [inline java/event-handlers/indicating-errors.md:155]:
```suggestion
For actions or functions, a message target can either be relative to the entity to which the action or function is bound (represented by the `cqn` parameter) or relative to a parameter of the action or function.
```

**Review by smahati — APPROVED** [review]

**Pull request merged**